pkg/netrunner/runner.go

```diff
@@ -4,16 +4,51 @@ import (
 	"log"
 	"net"
 	"net/http"
+	"regexp"
 	"strconv"
+	"sync"
 	"time"
 
 	"dish/pkg/config"
 	"dish/pkg/socket"
 )
 
+func TestSocket(sock socket.Socket, channel chan <- socket.Result, wg *sync.WaitGroup) {
```

Comment:
It's usually safer to return a channel you're going to write to instead of accepting one from the receiver - the sender should be responsible for the channel's lifecycle while the receiver should not be exposed to its implementation details.